src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -1773,6 +1773,21 @@ private bool FindAssociatedFiles()
             return newDependencies;
         }
 
+        // Targets that raise references from NuGet packages set FindDependencies=false to skip looking
+        // for dependencies that they have already resolved and related files that they are responsible
+        // for deploying.
+        private static bool ShouldFindDependencies(Reference reference)
```

Comment:
Should this be in Reference itself?

I was wondering if a more general name is suited. I want to experiment with this to turn off other RAR behaviour.

Reply:
Which other behaviours that is per-reference?

Reply:
When resolving a reference, RAR first looks near all the parents of that reference. If it finds nothing, then it starts asking the resolvers that were actually specified to RAR. I think this could be skipped when the reference is a nuget reference. For example, ProjA -> ProjB ->Nupkg. It's not worth looking for the Nupkg alongside ProjA or ProjB, and skip directly to the actual RAR resolvers. Though even in this case, the `FindDependencies` name is still valid.

Reply:
@cdmihai I like that optimization